src/diff_opt.jl

```diff
@@ -11,8 +11,55 @@
 
 const MOIDD = MOI.Utilities.DoubleDicts
 
+"""
+    LuFactorizationWithInertiaCorrection{T<:Real}
+
+A callable struct to store the parameters for the inertia correction in the
+Lu-factorization. If no inertia correction is needed, it only performs the LU
+factorization.
```

Comment:
needs comments about the parameters.

we need to think a bit more about this.

Initially I thought about having the parameters st and max corrections as atrrbitues.

Do we expect people will pass their factorization routines?

Reply:
This allows user to use better factorization routines like the ones in HSL.